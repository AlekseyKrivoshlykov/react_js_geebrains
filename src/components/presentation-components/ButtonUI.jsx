export default function ButtonUI(props) {

    return (
        <button {...props} className="btn btn-primary">Change visible</button>
    );
}