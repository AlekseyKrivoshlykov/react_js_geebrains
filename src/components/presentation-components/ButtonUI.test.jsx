import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import ButtonUI from './ButtonUI'

describe('Button UI', () => {
    it('button text', () => {
        render(<ButtonUI>Change visible</ButtonUI>)
        expect(screen.getByText(/Change visible/)).toBeInTheDocument()
    })

    it('render with snapshot', () => {
        const {asFragment} = render(<ButtonUI>Change visible</ButtonUI>)
        expect(asFragment()).toMatchSnapshot()
    })

    it('button have class btn-primary', () => {
        render(<ButtonUI>Change visible</ButtonUI>)
        expect(screen.getByText(/Change visible/)).toHaveClass('btn-primary')
    })

})




