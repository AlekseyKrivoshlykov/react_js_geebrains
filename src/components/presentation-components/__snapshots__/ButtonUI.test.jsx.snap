// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    Change visible
  </button>
</DocumentFragment>
`;

exports[`Button render render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    Change visible
  </button>
</DocumentFragment>
`;
